### PR TITLE
Add correct use for Error and io

### PR DESCRIPTION
**Pull request by lawrencewoodman**
This also repeats the case analysis used.

### CODE REVIEW

File: src/doc/book/error-handling.md [hunk 1]
```diff
@@ -1795,6 +1795,10 @@ To convert this to proper error handling, we need to do the following:
 Let's try it:
 
 ```rust,ignore
+use std::error::Error
```
Comment (durka) [inline src/doc/book/error-handling.md]: missing semicolon

**Comment by steveklabnik** [comment]
err actually new commit

@bors: r+ rollup

**Pull request merged**